src/app/dataElement/data-element/data-element.component.ts

```diff
@@ -221,7 +215,20 @@ export class DataElementComponent
         resource.minMultiplicity = item.minMultiplicity as number;
         resource.maxMultiplicity = item.maxMultiplicity;
       } else if (item.controlType === ProfileControlTypes.dataType) {
-        resource.dataType = item.value as DataTypeReference;
+        const castedValue = item.value as DataType;
```

Comment:
Not a necessary change but given we've identified that `item.controlType` is `datatype` a more descriptive name than `castedValue` might be `dataTypeValue`.